daemon/cluster/networks.go

```diff
@@ -87,9 +87,14 @@ func (c *Cluster) UpdateAttachment(target, containerID string, config *network.N
 	if !ok || attacher == nil {
 		return fmt.Errorf("could not find attacher for container %s to network %s", containerID, target)
 	}
+	if attacher.config != nil {
```

Comment:
That doesn't sound right to me either. Unless there's an upper bound on the number of times `UpdateAttachment` can be called, it would be possible to exhaust the buffer in the channel. What about having boolean that's set (with the lock held) when the first call to `UpdateAttachment` sends over the channel?